Objects/longobject.c

```diff
@@ -6212,6 +6212,25 @@ int___round___impl(PyObject *self, PyObject *o_ndigits)
         return long_long(self);
     }
 
+    int64_t n, e;
+
+    if (PyLong_AsInt64(ndigits, &n) < 0) {
+        Py_DECREF(ndigits);
+        return NULL;
+    }
+    n = -n;
+
+    /* A quick exit, if the result is zero.
+     * We start from criteria abs(self) < 10**(n - 1).  Then, from
+     * definition of the _PyLong_Frexp() we have
+     * abs(self) < 10*10**(log10(2)*e) < 10**(e/3 + 1). */
+    (void)_PyLong_Frexp((PyLongObject *)self, &e);
```

Comment:
This can return an error in theory. Do we care about it here?

Reply:
No, after d08c788822 - it's always successful.